ext/ldap/ldap.c

```diff
@@ -2573,31 +2573,31 @@ PHP_FUNCTION(ldap_modify_batch)
 
 		/* make sure the DN contains no NUL bytes */
 		if (_ldap_strlen_max(dn, dn_len) != dn_len) {
-			php_error_docref(NULL, E_WARNING, "DN must not contain NUL bytes");
-			RETURN_FALSE;
+			zend_argument_type_error(2, "cannot contain any null-bytes");
+			RETURN_THROWS();
 		}
 
 		/* make sure the top level is a normal array */
 		zend_hash_internal_pointer_reset(Z_ARRVAL_P(mods));
 		if (zend_hash_get_current_key_type(Z_ARRVAL_P(mods)) != HASH_KEY_IS_LONG) {
-			php_error_docref(NULL, E_WARNING, "Modifications array must not be string-indexed");
-			RETURN_FALSE;
+			zend_argument_type_error(3, "must be integer-indexed");
+			RETURN_THROWS();
 		}
 
 		num_mods = zend_hash_num_elements(Z_ARRVAL_P(mods));
 
 		for (i = 0; i < num_mods; i++) {
 			/* is the numbering consecutive? */
 			if ((fetched = zend_hash_index_find(Z_ARRVAL_P(mods), i)) == NULL) {
-				php_error_docref(NULL, E_WARNING, "Modifications array must have consecutive indices 0, 1, ...");
-				RETURN_FALSE;
+				zend_argument_value_error(3, "must have consecutive integer indices starting from 0");
+				RETURN_THROWS();
 			}
 			mod = fetched;
 
 			/* is it an array? */
 			if (Z_TYPE_P(mod) != IS_ARRAY) {
-				php_error_docref(NULL, E_WARNING, "Each entry of modifications array must be an array itself");
-				RETURN_FALSE;
+				zend_argument_value_error(3, "must contain only array items");
```

Comment:
```suggestion
				zend_argument_value_error(3, "must only contain arrays");
```
for consistency with the error message directly below, maybe?

Reply:
It's ok for me!